### DISCUSSION TITLE
Prompt for user to install update more frequently

### DISCUSSION THREAD

**Discussion by 011235813** [sparkle-project/Sparkle#2450] [General]
Currently, Sparkle has the option to check for updates periodically (e.g. every day) and download updates in the background. But it doesn't auto-install if the app is open, since that requires the user to quit and re-open the app. However, some users don't quit the app for a very long time, e.g. more than a week. So, the user would not know that an update is available. From this [post](https://github.com/sparkle-project/Sparkle/issues/1576), it looks like Sparkle has something called `SUImpatientUpdateCheckInterval` that notifies users of an update. It seems this is set to a default of a week (is this documented anywhere?). Can this `SUImpatientUpdateCheckInterval` be changed via the Info.plist? If so, what are the units (e.g. seconds, hours, days, weeks)?

**Comment by zorgiepoo** [comment]
It is in seconds and not currently configurable. If it was configurable, what would you want to set it to? Or why isn't 1 week a good default for you?

It is intended to be long enough where a lot of users may quit the app before the impatient interval check is hit and not need to be notified, but short enough that it gets hit for users that keep an app running for a 'very long time'.

**Reply by 011235813** [reply]
Thanks for the reply!

> If it was configurable, what would you want to set it to? 

I may want to set it differently based on the stage of app deployment, e.g. shorter at the start when pushing patches frequently, longer later. Making it configurable is also useful for testing that it works at all, as currently it's not feasible for me to have an app open on some other machine for one week to test that the notification works.

**Reply by zorgiepoo** [reply]
Feel free to open a PR to make this configurable in the Info.plist if you find this actually useful. I don't want it to be configurable by user defaults.

Testability is a different story as regular update checking is bound by minimum of 1 hour, and I believe impatient interval is bound by the regular update checking interval. So in a Release build of Sparkle you'd potentially have to wait 2 hours to test it. Debug builds of Sparkle have much less limits (1 minute and 2 minutes respectively) but many developers do not use debug builds of Sparkle (we don't ship them in any of the package managers for example). One day it may be nice to remove this debug/release separation and have some private Info.plist debug mode key instead..

**Reply by zorgiepoo** [reply]
One more thing. You may also want to check out [-updater:willInstallUpdateOnQuit:immediateInstallationBlock:](https://sparkle-project.org/documentation/api-reference/Protocols/SPUUpdaterDelegate.html#/c:objc(pl)SPUUpdaterDelegate(im)updater:willInstallUpdateOnQuit:immediateInstallationBlock:) which some developers use instead (delegate is responsible for handling immediate installation of update instead of Sparkle)

**Reply by 011235813** [reply]
> Testability is a different story as regular update checking is bound by minimum of 1 hour, and I believe impatient interval is bound by the regular update checking interval. So in a Release build of Sparkle you'd potentially have to wait 2 hours to test it.

Sounds good, 2hrs is reasonable for testing. So we just need to get `SUImpatientUpdateCheckInterval` added to Info.plist.
